packages/2d/src/components/Icon.ts

```diff
@@ -0,0 +1,130 @@
+   * * ph:activity-bold
+   */
+  icon: SignalValue<string>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  color?: SignalValue<string>;
+}
+
+/**
+ * An Icon Component that provides an easy access to over 150k icons.
+ * See https://icones.js.org for all available Icons.
+ */
+export class Icon extends Image {
+  /**
+   * Internal cache
+   */
+  private static iconContentsPool: Record<string, HTMLImageElement> = {};
+
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  @signal()
+  public declare icon: SimpleSignal<string, this>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  @initial('white')
+  @signal()
+  public declare color: SimpleSignal<string, this>;
+
+  public constructor(props: IconProps) {
+    super(props);
+  }
+
+  /**
+   * This creates the URL that will be used as the Image source
+   * @returns Address to Iconify API for the requested Icon.
+   */
+  @computed()
+  protected svgUrl(): string {
+    const iconPathSegment = this.icon().replace(':', '/');
+
+    // iconify API is documented here: https://docs.iconify.design/api/svg.html#color
+    const url = `https://api.iconify.design/${iconPathSegment}.svg?color=${Icon.normalizeColorInput(
+      this.color(),
+    )}`;
+    return url;
+  }
+
+  private static normalizeColorInput(colorString: string) {
+    // This assumes that no HTML named color is a valid hexadecimal string.
+    // First, check if a # is present. If yes, we can assume that this is not a
+    // named color.
+    const hasHashtag = colorString.trimStart().startsWith('#');
+
+    // Does no longer contain a #. Also no trailing / leading whitespace characters
+    const withoutHashTag = hasHashtag
+      ? colorString.split('#')[1].trim()
+      : colorString.trim();
+
+    function canBeInterpretedAsHexadecimal(str: string) {
+      if (str.length !== 6 && str.length !== 8) {
+        return false;
+      }
+      // If we do not have a hexadecimal string here, parseInt will return Not a Number (NaN)
+      // so we have a valid hexadecimal string if this returns false
+      return !Number.isNaN(parseInt(str, 16));
+    }
+
+    const colorName = canBeInterpretedAsHexadecimal(withoutHashTag)
+      ? '%23' + withoutHashTag // as hexadecimal value
```

Comment:
You might replace `%23` with `encodeURIComponent('#')` just so it's clear where this string is coming from.